guardian/editions
# remove hasOnboarded and rely on privacy settings instead

## Summary

People would systematically get the onboarding again and again on startup if they pressed "See Options" instead of "I'm okay with that" (enabling all flags to `true`). This is because we wouldn't set `hasOnboarded` to `true` in that code path.

All in all, onboarding is directly connected to whether we make an explicit choice on GDPR buckets or not. So we're agreed to remove the `hasOnboarded` flag with @blishen and instead just rely on presence or not of the GDPR flags.

## Test Plan

1. Uninstall app, reinstall from fresh.
2. Open app, notice onboarding screen. Press See Options, "accept all".
3. Close app, reopen, notice there is no onboarding again (broken without this changeset).
4. Uninstall app, reinstall from fresh.
5. Open app, press "I'm okay with that". Close app, reopen, notice there is no onboarding again.
